### DISCUSSION TITLE
Is the string ".php" stripped from short urls?

### DISCUSSION THREAD

**Discussion by markjr** [YOURLS/YOURLS#3502] [General]
Hi, I am trying to set up a private instance of Yourls to use for fixing old backlinks to a few hostnames we no longer use (I want to retain their mappings to the appropriate pages on our site now, rather than just 301 the entire domain to our homepage).

I'm getting 403 errors, I know what those are about (haven't put in a lander page there yet) - but not always.

At first I thought it was something weird between having multiple plugins enabled for:

- allow hyphens
- allow full stops
- allow forward slashes

(and I also created an allow underscores which is just a copy of allow hyphens, modified)

It sometimes worked, sometimes didn't as I was trying all manner of combos above

Until I realized that when I was trying to set up a short url that ended in '.php' is when I get the problem - always.

Is this being stripped out as some kind of safety / sanity mechanism and if so can I work around that? 

As you can imagine, grepping the code for \.php brings up a lot of stuff.

Thx, mark

**Comment by dgw** [comment]
The `.php` suffix almost certainly triggers a handler on your server that runs before YOURLS.

**Reply by markjr** [reply]
> The `.php` suffix almost certainly triggers a handler on your server that runs before YOURLS.

I get that - but unless I misunderstand how Yourls works, isn't it in this case just a value in a database that gets selected on? 
The webserver wouldn't even really see it.

**Reply by markjr** [reply]
You're saying the webserver sees the incoming request for /something.php before the .htaccess that loads yourls-loader.php even gets executed?

**Reply by dgw** [reply]
Yes, that's exactly what I'm saying. It sees a request for a PHP file that doesn't exist, and returns 403.

Check higher levels of your Apache configuration to see if you can add an "is file" check to the PHP logic. If YOURLS runs first, its htaccess snippet skips looking up a short URL if the requested path exists on the file system. You want the PHP logic to be skipped if the request path does _not_ exist on the file system.

**Reply by markjr** [reply]
I was just wondering if an ErrorDoc 404 loading /yourls-loader.php would do the trick

(I think my guys used nginx so I'm even more clueless than usual)

**Comment by markjr** [comment]
We switched to Apache and one of my guys figured it out, so it works now.

Thanks for your help, I should have realized I was looking in the wrong place.
